tests/test_ide_integration.py

```diff
@@ -0,0 +1,112 @@
+# This require loding leap ide docker image and run inside that image
+class LeapIdeIntegrationTests(unittest.TestCase):
+
+    def setUp(self):
+        self.verificationErrors = []
+        self.logger = None
+        self.log = self.custom_logger(logging.DEBUG)
+
+    def custom_logger(self,log_level=logging.DEBUG):
+
+        # Check if logger object exists
+        if self.logger is not None:
+            return self.logger
+
+        # Gets the name of the class / method from where this method is called
+        logger_name = inspect.stack()[1][3]
+        self.logger = logging.getLogger(logger_name)
+
+        # By default, log all messages
+        self.logger.setLevel(logging.DEBUG)
+
+        # Get log file name from environment getting set in testRunner.sh
+        log_file= os.getcwd()+"logs.log"
+        if os.getenv('log_file') is not None and os.getenv('log_file'):
+            log_file = os.environ['log_file']
+
+        # File handler for logging
+        file_handler = logging.FileHandler(log_file, mode='a')
+        file_handler.setLevel(log_level)
+        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s: %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+        file_handler.setFormatter(formatter)
+        self.logger.addHandler(file_handler)
+
+        # Log to console
+        console_handler = logging.StreamHandler()
+        console_handler.setFormatter(formatter)
+        console_handler.setLevel(log_level)
+        self.logger.addHandler(console_handler)
+
+        return self.logger
+
+    def VerifyErrors(self,output):
+
+        try: 
+            self.assertNotIn("ERROR",output.upper() )
+            self.log.info("Verify if error string contains in output successful \n")
+        except AssertionError as e:
+            self.log.error("Verify if error string contains in output failed \n")
+            self.verificationErrors.append(str(e))
+
+        try: 
+            self.assertNotIn("WARNING",output.upper() )
+            self.log.info("Verify if warning string contains in output successful \n")
+        except AssertionError as e:
+            self.log.error("Verify if warning string contains in output failed \n")
+            self.verificationErrors.append(str(e))
+
+    def test_factoring(self):
+        cwd=os.getcwd()
+        # os.chdir(cwd+"")
+        p = Popen(["python",  cwd+"/demo.py"], stdout=PIPE, stdin=PIPE, stderr=STDOUT)    
+        p.stdin.write(b'49\n')
+        time.sleep(5)
+        output = p.communicate()[0]
+        output=str(output)
+        self.log.info("Example output \n"+output)
+        os.chdir(cwd)
+
+        try: 
+            self.assertIn("'results': [{'a': 7,".upper(),output.upper() )
+            self.log.info("Test factoring example verification successful \n")
+        except AssertionError as e:
+            self.log.error("Test factoring example verification failed \n")
+            self.verificationErrors.append(str(e))
```

Comment:
Same here, it's unnecessary to have a try-except when you're using `unittest`'s `assertIn`

Reply:
@m3ller @randomir I added this as i want to continue execution for next validation if even if one verification failed , good know which next validation pass
If you remove try catch , it will stop execution and will not proceed to verify next one

let me know if this make sence to you or you have better approch